File: capture.js
// 粘 URL → 抓 <title>+meta 描述 → 追加一行进 inbox/inbox.md。
// 用法: node tools/capture.js <URL> [备注]   (python 版见 capture.py;本机 python 是商店空壳,用这个)
const https = require('https'), http = require('http'), fs = require('fs'), path = require('path');
const INBOX = path.join(__dirname, '..', 'inbox', 'inbox.md');
const url = process.argv[2];
const note = process.argv.slice(3).join(' ').trim();
if (!url) { console.log('用法: node tools/capture.js <URL> [备注]'); process.exit(1); }
function unescape(s){ return s.replace(/&amp;/g,'&').replace(/&lt;/g,'<').replace(/&gt;/g,'>').replace(/&quot;/g,'"').replace(/&#39;/g,"'").replace(/\s+/g,' ').trim(); }
function fetch(u, cb, redirects=0){
  const lib = u.startsWith('http://') ? http : https;
  const req = lib.get(u, {headers:{'User-Agent':'Mozilla/5.0'}, rejectUnauthorized:false, timeout:30000}, res=>{
    if ([301,302,303,307,308].includes(res.statusCode) && res.headers.location && redirects<5){
      res.resume(); return fetch(new URL(res.headers.location, u).href, cb, redirects+1);
    }
    let data=''; res.setEncoding('utf8'); res.on('data',d=>data+=d); res.on('end',()=>cb(null,data));
  });
  req.on('error', e=>cb(e)); req.on('timeout', ()=>{ req.destroy(); cb(new Error('timeout')); });
}
fetch(url, (err, body)=>{
  let title='', desc='';
  if (!err && body){
    const mt = body.match(/<title[^>]*>([\s\S]*?)<\/title>/i); if (mt) title = unescape(mt[1]);
    const md = body.match(/<meta[^>]+name=["']description["'][^>]+content=["']([\s\S]*?)["']/i); if (md) desc = unescape(md[1]);
  }
  const tag = err ? `(抓取失败:${err.message})` : (note || title || desc || '(无标题)');
  const line = `- [${new Date().toISOString().slice(0,10)}] ${url} — ${tag}`;
  fs.appendFileSync(INBOX, '\n'+line+'\n', 'utf8');
  console.log('[captured]', line);
});

File: capture.py
import datetime, html, os, re, ssl, sys, urllib.request
WIKI_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
INBOX = os.path.join(WIKI_ROOT, "inbox", "inbox.md")
try: sys.stdout.reconfigure(encoding="utf-8")
except Exception: pass
def fetch_meta(url):
    ctx = ssl.create_default_context(); ctx.check_hostname=False; ctx.verify_mode=ssl.CERT_NONE
    raw = urllib.request.urlopen(urllib.request.Request(url, headers={"User-Agent":"Mozilla/5.0"}), timeout=30, context=ctx).read()
    enc="utf-8"; m=re.search(rb'charset=["\']?([\w\-]+)', raw[:4000], re.I)
    if m:
        try: enc=m.group(1).decode("ascii")
        except Exception: pass
    t=raw.decode(enc, errors="ignore")
    title=""; mt=re.search(r"<title[^>]*>(.*?)</title>", t, re.I|re.S)
    if mt: title=html.unescape(" ".join(mt.group(1).split()))
    desc=""; md=re.search(r'<meta[^>]+name=["\']description["\'][^>]+content=["\'](.*?)["\']', t, re.I|re.S)
    if md: desc=html.unescape(" ".join(md.group(1).split()))
    return title, desc
def main():
    if len(sys.argv)<2: print("用法: python tools/capture.py <URL> [备注]"); sys.exit(1)
    url=sys.argv[1]; note=" ".join(sys.argv[2:]).strip()
    try: title,desc=fetch_meta(url)
    except Exception as e: title,desc="",f"(抓取失败:{e})"
    line=f"- [{datetime.datetime.now():%Y-%m-%d}] {url} — {note or title or desc or '(无标题)'}"
    open(INBOX,"a",encoding="utf-8").write("\n"+line+"\n")
    print("[captured]", line)
if __name__=="__main__": main()
